std/algorithm/comparison.d

```diff
@@ -610,13 +610,26 @@ if (isInputRange!R1 && isInputRange!R2)
 {
     static if (!(isSomeString!R1 && isSomeString!R2))
     {
+        static if (is(typeof(pred) : string))
+            enum isLessThan = pred == "a < b";
```

Comment:
FYI: @andralex isn't a huge fan of string comparisons, see https://github.com/dlang/phobos/pull/4265#issuecomment-266234691

Reply:
As I did a bit of digging the string specialization of `cmp` was added in 2011: https://github.com/dlang/phobos/commit/a0ecf2a10e319e387e269fd6310ad2acd244cc92

Reply:
Good news, https://github.com/dlang/dmd/pull/7484/ is on. With it we should be able to switch to lambdas soon. cc @RazvanN7